sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -459,12 +460,24 @@ private Uni<Integer> determineTtl(
         .getDocumentTtlInfo(namespace, collection, documentId, ctx)
         .onItem()
         .transform(
-            rawDocument ->
-                rawDocument
-                    .rows()
-                    .get(0)
-                    .getLong(String.format("ttl(%s)", configuration.table().leafColumnName()))
-                    .intValue())
+            rawDocument -> {
+              if (rawDocument == null) {
+                return 0;
+              }
```

Comment:
can we replace this with

`.onItem().IfNotNull()`

you are handling the null case at the end